### PR TITLE
#35568 Zero warnings

### PR DESCRIPTION
**Pull request by svick**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -4,6 +4,7 @@
         <RepoDirectory>$(MSBuildThisFileDirectory)</RepoDirectory>
         <RepoKind>GitHub</RepoKind>
         <DisableCodeQuality>True</DisableCodeQuality>
+        <NoWarn>$(NoWarn);NU1903;NU1904</NoWarn>
```
Comment (prochan2) [inline Directory.Build.props:9]: Comment on what these warnings mean, why they are suppressed, and when (if ever) this suppression should be removed.

File: src/PostSharp.Engineering.BuildTools/Build/Model/ArtifactPublisher.cs [hunk 2]
```diff
@@ -44,18 +36,18 @@ protected override bool Publish(
 
             var directory = isPublic ? directories.Public : directories.Private;
 
-            var files = new List<FilePatternMatch>();
+            var files1 = new List<FilePatternMatch>();
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Build/Model/ArtifactPublisher.cs]:
```suggestion
            var matchedFiles = new List<FilePatternMatch>();
```

File: src/PostSharp.Engineering.BuildTools/Build/Model/ArtifactPublisher.cs [hunk 3]
```diff
@@ -8,28 +8,20 @@
...
     /// <summary>
-    /// A publisher that publishes all artifact files specified in <see cref="Files"/> pattern.
+    /// A publisher that publishes all artifact files specified in <c>files</c> pattern.
     /// </summary>
-    public abstract class ArtifactPublisher : Publisher
+    public abstract class ArtifactPublisher( Pattern files ) : Publisher
     {
-        public Pattern Files { get; }
-
-        public Tester[] Testers { get; init; } = [];
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Build/Model/ArtifactPublisher.cs]: This has to stay public. See Metalama.Try, for example.

File: src/PostSharp.Engineering.BuildTools/Build/Model/Product.cs [hunk 4]
```diff
@@ -2026,7 +2026,7 @@ public bool Swap( BuildContext context, SwapSettings settings )
                         case SuccessCode.Success:
                             foreach ( var tester in swapper.Testers )
                             {
-                                switch ( tester.Execute( context, directories.Private, buildInfo, configuration, settings.Dry ) )
+                                switch ( tester.Execute( context, directories.Private, buildInfo, settings.Dry ) )
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Build/Model/Product.cs:2061]: CHECK

File: src/PostSharp.Engineering.BuildTools/ContinuousIntegration/TeamCityCreateProjectCommandSettings.cs [hunk 5]
```diff
@@ -1,10 +1,12 @@
 ﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
 
+using JetBrains.Annotations;
 using Spectre.Console.Cli;
 using System.ComponentModel;
 
 namespace PostSharp.Engineering.BuildTools.ContinuousIntegration;
 
+[PublicAPI]
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/ContinuousIntegration/TeamCityCreateProjectCommandSettings.cs]: [UsedImplicitly] should be user for *CommandSettings as the reason for them being public is that the Sprectre.Console instantiates them.

File: src/PostSharp.Engineering.BuildTools/Dependencies/Model/BuildOrderHelper.cs [hunk 6]
```diff
@@ -1,14 +0,0 @@
-// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
-
-namespace PostSharp.Engineering.BuildTools.Dependencies.Model;
-
-public static class BuildOrderHelper
-{
-    public static void SetBuildOrder( params DependencyDefinition[] orderedProducts )
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Dependencies/Model/BuildOrderHelper.cs:7]: CHECK

File: src/PostSharp.Engineering.BuildTools/Git/GitBulkRenameCommand.cs [hunk 7]
```diff
@@ -1,20 +1,21 @@
...
 {
     /// <summary>
     /// Renames files in bulk using <c>git rename</c>.
     /// </summary>
+    [UsedImplicitly]
     internal class GitBulkRenameCommand : Command<GitBulkRenameSettings>
     {
         // ReSharper disable RedundantNullableFlowAttribute
-        public override int Execute( [NotNull] CommandContext context, [NotNull] GitBulkRenameSettings settings )
+        public override int Execute( [System.Diagnostics.CodeAnalysis.NotNull] CommandContext context, [System.Diagnostics.CodeAnalysis.NotNull] GitBulkRenameSettings settings )
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Git/GitBulkRenameCommand.cs]: Why?

File: src/PostSharp.Engineering.BuildTools/NuGet/RenamePackagesCommand.cs [hunk 8]
```diff
@@ -10,21 +11,18 @@
...
     public class RenamePackagesCommand : Command<RenamePackageCommandSettings>
     {
         public override int Execute( CommandContext context, RenamePackageCommandSettings settings )
-        {
-            return Execute( new ConsoleHelper(), settings ) ? 0 : 2;
-        }
+            => Execute( new ConsoleHelper(), settings ) ? 0 : 2;
 
-        public static bool Execute( ConsoleHelper console, RenamePackageCommandSettings settings )
+        private static bool Execute( ConsoleHelper console, RenamePackageCommandSettings settings )
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/NuGet/RenamePackagesCommand.cs]: This is used in Metalama Compiler's engineering.

File: src/PostSharp.Engineering.BuildTools/Utilities/DotNetTool.cs [hunk 9]
```diff
@@ -21,7 +21,7 @@ public class DotNetTool
 
         public static DotNetTool SignClient { get; } = new SignTool();
 
-        public static DotNetTool Resharper { get; } = new( "jb", "JetBrains.Resharper.GlobalTools", "2023.3.4", "jb" );
+        public static DotNetTool Resharper { get; } = new( "jb", "JetBrains.Resharper.GlobalTools", "2024.2.7", "jb" );
```
Comment (prochan2) [inline src/PostSharp.Engineering.BuildTools/Utilities/DotNetTool.cs]: Add a comment on why this version is chosen / why it cannot be upgraded.

**Review by prochan2 — CHANGES_REQUESTED** [review]

File: src/PostSharp.Engineering.BuildTools/Utilities/DotNetTool.cs (hunk 9 above)
Comment (svick) [reply src/PostSharp.Engineering.BuildTools/Utilities/DotNetTool.cs:24]: I think this was the highest version when I made this change.
Should I upgrade to the newer one?

**Review by prochan2 — APPROVED** [review]

**Pull request merged**